### PR TITLE
DOC: document actions

### PR DESCRIPTION
**Pull request by jorgepiloto**
This pull-request documents the different reusable actions available. Information about the inputs and code snippets are provided in the `README.rst` file to guide users.

### CODE REVIEW

**Comment by jorgepiloto** [comment]
I thought about creating a Sphinx-based documentation for this. Maybe, this would be interesting as we could test our workflows. Only the `release-private` and `release-public` would be untested.

File: doc/source/build-actions/index.rst
Comment (Revathyvenugopal162) [inline doc/source/build-actions/index.rst:34]:
```suggestion
C-extension.
```

File: doc/source/build-actions/index.rst
Comment (Revathyvenugopal162) [inline doc/source/build-actions/index.rst:45]:
```suggestion
      runs-on: ${{ matrix.os }}
      strategy:
         matrix:
             os: [ubuntu-latest, windows-latest, macos-11]
```
Just an example that the user can build wheels for any os. But adding the matrix in example or not is up to your call.

File: doc/source/conf.py
Comment (Revathyvenugopal162) [inline doc/source/conf.py:6]:
```suggestion
from ansys_sphinx_theme import ansys_favicon, ansys_logo_black
```

File: doc/source/conf.py
Comment (Revathyvenugopal162) [inline doc/source/conf.py:23]:
```suggestion
html_theme = "ansys_sphinx_theme"
html_favicon = ansys_favicon
```

File: doc/source/style-actions/index.rst
Comment (Revathyvenugopal162) [inline doc/source/style-actions/index.rst:14]:
```suggestion
``.pre-commit-config.yaml`` file in the root directory.
```

File: requirements/requirements_doc.txt
Comment (Revathyvenugopal162) [inline requirements/requirements_doc.txt:2]:
```suggestion
ansys-sphinx-theme==0.7.0
```

**Comment by jorgepiloto** [comment]
Pinging here @PipKat. For some reason, GitHub is not allowing me to add you as a reviewer, Kathy.

**Comment by Revathyvenugopal162** [comment]
> Pinging here @PipKat. For some reason, GitHub is not allowing me to add you as a reviewer, Kathy.

I think you have to add Documentation team in settings/access/collaborators and team and give write access.

**Comment by jorgepiloto** [comment]
You are right, @Revathyvenugopal162. I just added documentation team to this repository and requested a review by Kathy.

File: .github/workflows/ci_cd.yml
Comment (PipKat) [inline .github/workflows/ci_cd.yml:97]: @jorgepiloto Set up (2 words) when used as a verb and setup (1 word) when used as a noun.

File: .gitignore
Comment (PipKat) [inline .gitignore:91]:
```suggestion
#   For a library or package, you might want to ignore these files because the code is
```

File: .gitignore
Comment (PipKat) [inline .gitignore:92]:
```suggestion
#   intended to run in multiple environments. Otherwise, check them in:
```
Your usage is correct. It's just that Google style guidelines say to avoid semicolons because not all users interpret them correctly.

File: .gitignore
Comment (PipKat) [inline .gitignore:96]:
```suggestion
#   According to pypa/pipenv#598, you should include Pipfile.lock in version control.
```

File: .gitignore
Comment (PipKat) [inline .gitignore:100]: Is a blank line needed between lines 99 and 100?

File: .gitignore
Comment (PipKat) [inline .gitignore:103]:
```suggestion
#   Similar to the Pipfile.lock recommendation, you should generally include poetry.lock in version control.
```

File: .gitignore
Comment (PipKat) [inline .gitignore:105]:
```suggestion
#   This is especially recommended for binary packages to ensure reproducibility. This
#   recommendation is commonly ignored for libraries.
```

File: .gitignore
Comment (PipKat) [inline .gitignore:156]:
```suggestion
#  A JetBrains-specific template is maintained in a separate JetBrains.gitignore here:
#  https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore
#  You can add this template to the global gitignore or merge it into this file.  For a more nuclear
#  option (not recommended), you can uncomment the following to ignore the entire idea folder.
```

File: README.rst
Comment (PipKat) [inline README.rst:18]:
```suggestion
For more information on available actions and how to use them, see `actions.docs.pyansys.com <https://actions.docs.pyansys.com>`_ .
```
Style guidelines say to start with why they would go see a resource and to use "on" as the preposition.

File: doc-build/action.yml
Comment (PipKat) [inline doc-build/action.yml:11]:
```suggestion
    description: 'Set of options to pass to the Sphinx builder'
```

File: doc-style/action.yml
Comment (PipKat) [inline doc-style/action.yml:7]: I prefer to use of the double quotations--mainly because that is what Alex told me to use when I did the overall edit Of PyAEDT before its public debut.

File: doc/.vale.ini
Comment (PipKat) [inline doc/.vale.ini:7]:
```suggestion
# The options are ``"suggestion"``, ``"warning"``, and ``"error"``. The default is ``“warning”``.
```

Comment (PipKat) [reply doc/.vale.ini:7]: This might be open to debate as the subsequent lines are just using a single back tick. Do they differ in what they mean? We should be consistent.

File: doc/.vale.ini
Comment (PipKat) [inline doc/.vale.ini:16]:
```suggestion
# WordTemplate specifies what Vale considers to be an individual word.
```

File: doc/.vale.ini
Comment (PipKat) [inline doc/.vale.ini:19]:
```suggestion
# List of packages to use for our guidelines
```

File: doc/Makefile
Comment (PipKat) [inline doc/Makefile:4]:
```suggestion
# You can set these variables from the command line and also
```

File: doc/Makefile
Comment (PipKat) [inline doc/Makefile:5]:
```suggestion
# You can set these variables from the command line. You can also
# set the first two from the environment.
```

File: doc/make.bat
Comment (PipKat) [inline doc/make.bat:19]:
```suggestion
	echo.may add the Sphinx directory to the PATH environment variable.

```

File: doc/make.bat
Comment (PipKat) [inline doc/make.bat:21]:
```suggestion
	echo.If you don't have Sphinx installed, you can obtain it from

```

File: doc/source/build-actions/index.rst
Comment (PipKat) [inline doc/source/build-actions/index.rst:10]:
```suggestion
This action builds source and wheel artifacts for a Python library.
```

File: doc/source/build-actions/index.rst
Comment (PipKat) [inline doc/source/build-actions/index.rst:17]:
```suggestion
Here is a code sample for using this action:
```

File: doc/source/build-actions/index.rst
Comment (PipKat) [inline doc/source/build-actions/index.rst:33]:
```suggestion
This action builds wheel artifacts for a Python library using
```
Wondering if C-extension is a name that should be in a literal tag or perhaps be a link?

File: doc/source/conf.py
Comment (PipKat) [inline doc/source/conf.py:21]:
```suggestion
# Use the default pyansys logo
```

File: doc/source/conf.py
Comment (PipKat) [inline doc/source/conf.py:26]:
```suggestion
# Specify the location of your GitHub repo
```

File: doc/source/conf.py
Comment (PipKat) [inline doc/source/conf.py:38]:
```suggestion
# Define Sphinx extensions
```

File: doc/source/conf.py
Comment (PipKat) [inline doc/source/conf.py:47]:
```suggestion
# Specify the static path
```

File: doc/source/conf.py
Comment (PipKat) [inline doc/source/conf.py:50]:
```suggestion
# Add any paths that contain templates, relative to this directory.
```

File: doc/source/conf.py
Comment (PipKat) [inline doc/source/conf.py:53]:
```suggestion
# Specify the suffixes of source filenames.
```

File: doc/source/conf.py
Comment (PipKat) [inline doc/source/conf.py:56]:
```suggestion
# Specify the master toctree document
```

File: doc/source/conf.py
Comment (PipKat) [inline doc/source/conf.py:59]:
```suggestion
# Generate section labels for up to four levels
```

File: doc/source/conf.py
Comment (PipKat) [inline doc/source/conf.py:64]:
```suggestion
    """Read and convert input variables of a GitHub YAML action file into a RST table.
```

File: doc/source/conf.py
Comment (PipKat) [inline doc/source/conf.py:69]:
```suggestion
        Name of the action file.
```

File: doc/source/conf.py
Comment (PipKat) [inline doc/source/conf.py:74]:
```suggestion
        String representing the RST table.
```

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:4]:
```suggestion
This action builds and deploys the documentation of
```

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:8]:
```suggestion
To use these actions, a project must use `Sphinx <https://www.sphinx-doc.org/en/master/>`_
as documentation parser.
```

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:17]:
```suggestion
It uses ``HTML``, ``PDF``, and ``JSON`` builders to generate the following
```

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:22]:
```suggestion
* ``documentation-html``: Web-based documentation to display in ``gh-pages``.
* ``documentation-pdf``: File-based documentation to use in offline tasks.
* ``documetation-json``: Documentation that is to be consumed by the Ansys developer's portal.
```

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:28]:
```suggestion
Here is a code sample for using this action:
```

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:45]:
```suggestion
This action deploys ``HTML`` documentation into the ``gh-pages`` branch. It is
expected to be used after the :ref:`Doc build action` because it looks for an
artifact named ``documentation-html``.
```

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:51]:
```suggestion
Here is a code sample for using this action:
```

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:70]:
```suggestion
This action deploys ``HTML`` documentation into the ``release/`` directory of
```

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:72]:
```suggestion
action` because it looks for an artifact named ``documentation-html``.
```

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:76]:
```suggestion
``<MAJOR>.<MINOR>``. If this directory already exists, its content is overridden.
```

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:78]:
```suggestion
The ``release/`` directory is expected to contain a JSON file for version
```

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:79]:
```suggestion
mapping, as specified in `Version switcher dropdowns
```
Google style guide says to match the heading for the linked topic.

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:81]:
```suggestion
in the `PyData Theme
```

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:82]:
```suggestion
<https://pydata-sphinx-theme.readthedocs.io/en/stable/index.html>`_ for Sphinx output.
```

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:85]:
```suggestion
values inside the ``html_theme_options`` element:
```
Goggle style says a noun has to follow a literal tagging so that a reader understands what it is or belongs to.

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:98]:
```suggestion
All previous logic supports multi-version documentation history in
a PyAnsys project.
```
I'm guessing her what you meant.

File: doc/source/doc-actions/index.rst
Comment (PipKat) [inline doc/source/doc-actions/index.rst:104]:
```suggestion
Here is a code sample for using this action:
```

File: doc/source/index.rst
Comment (PipKat) [inline doc/source/index.rst:2]: An introductory sentence should be added here.

File: doc/source/release-actions/index.rst
Comment (PipKat) [inline doc/source/release-actions/index.rst:3]:
```suggestion
Release actions provide for releasing the various artifacts of a Python library.
```

File: doc/source/release-actions/index.rst
Comment (PipKat) [inline doc/source/release-actions/index.rst:5]:
```suggestion
These actions assume that you have used the :ref:`Documentation actions`
```

File: doc/source/release-actions/index.rst
Comment (PipKat) [inline doc/source/release-actions/index.rst:12]:
```suggestion
This action deploys all Python library artifacts into the `Ansys
```

File: doc/source/release-actions/index.rst
Comment (PipKat) [inline doc/source/release-actions/index.rst:24]:
```suggestion
Here is a code sample for using this action:
```

File: doc/source/release-actions/index.rst
Comment (PipKat) [inline doc/source/release-actions/index.rst:43]:
```suggestion
This action deploys all Python library artifacts into the public
```

File: doc/source/release-actions/index.rst
Comment (PipKat) [inline doc/source/release-actions/index.rst:53]:
```suggestion
Here is a code sample for using this action:
```

File: doc/source/release-actions/index.rst
Comment (PipKat) [inline doc/source/release-actions/index.rst:72]:
```suggestion
This action deploys all Python library artifacts into the `GitHub
```

File: doc/source/release-actions/index.rst
Comment (PipKat) [inline doc/source/release-actions/index.rst:81]:
```suggestion
Here is a code sample for using this action:
```

File: doc/source/style-actions/index.rst
Comment (PipKat) [inline doc/source/style-actions/index.rst:3]:
```suggestion
Style actions check code and documentation quality compliance
```

File: doc/source/style-actions/index.rst
Comment (PipKat) [inline doc/source/style-actions/index.rst:6]:
```suggestion
To use these actions, a project must use `pre-commit
```

File: doc/source/style-actions/index.rst
Comment (PipKat) [inline doc/source/style-actions/index.rst:8]:
```suggestion
<https://pre-commit.com>`_ and `Vale <https://vale.sh>`_. For help
implementing these tools, send an email to `pyansys.support@ansys.com
```

File: doc/source/style-actions/index.rst
Comment (PipKat) [inline doc/source/style-actions/index.rst:20]:
```suggestion
Here is a code sample for using this action:
```

File: doc/source/style-actions/index.rst
Comment (PipKat) [inline doc/source/style-actions/index.rst:48]:
```suggestion
Here is a code sample for using this action:
```

**Review by PipKat — APPROVED** [review]
Here are my review comments. Lots of nitpicks.

File: .github/workflows/ci_cd.yml
Comment (jorgepiloto) [reply .github/workflows/ci_cd.yml:97]: Oh! Thanks a lot for clarifying this, @PipKat. I will apply it from now on.

File: .gitignore
Comment (jorgepiloto) [reply .gitignore:92]: Thanks for the nitpick. I will avoid their usage in our documentation then.

File: .gitignore
Comment (jorgepiloto) [reply .gitignore:156]: In this context, "this file" refers to the ".gitignore" file.

File: README.rst
Comment (jorgepiloto) [reply README.rst:18]: Taking notes on this, thanks for remembering this, Kathy.

File: doc-style/action.yml
Comment (jorgepiloto) [reply doc-style/action.yml:7]: Sure, it should be okay to apply those here. Let me remove the changes in this line.

The syntax for GitHub actions can be tricky. Some parameters need to use single quotes. Otherwise, they are badly interpreted. As said, I bet this one can make use of double quotes.

File: doc/.vale.ini
Comment (jorgepiloto) [reply doc/.vale.ini:7]: This is the configuration file for Vale. Thus, it is not expected to be rendered in our docs. I just copy-pasted what we have in other projects.

I guess we should discard these changes so the whole configuration file is uniform and similar to other config files across projects.

File: doc/source/build-actions/index.rst
Comment (jorgepiloto) [reply doc/source/build-actions/index.rst:33]: I was looking in the [C extension official documentation](https://docs.python.org/3/extending/extending.html) an noticed that:

* It should read `C extension` and not `C-extension`

Does this affect the usage of literal tag?

File: doc/source/doc-actions/index.rst
Comment (jorgepiloto) [reply doc/source/doc-actions/index.rst:17]: Sorry for missing that Oxford comma.

File: doc/source/doc-actions/index.rst
Comment (jorgepiloto) [reply doc/source/doc-actions/index.rst:85]: Understood, will apply from now on. I really need to read Google style guide in more detail.

File: doc/source/release-actions/index.rst
Comment (jorgepiloto) [reply doc/source/release-actions/index.rst:5]: I guess these words were removed because they are redundant, right?

File: doc-style/action.yml
Comment (jorgepiloto) [inline doc-style/action.yml:7]:
```suggestion
    default: "doc/.vale.ini"
```

File: doc/source/index.rst
Comment (jorgepiloto) [reply doc/source/index.rst:2]: The introduction here is the one in the `README.md` file, as this file gets included directly into the documentation.

Nevertheless, I think the introduction in the `README.md` file can be improved.

File: .github/workflows/ci_cd.yml
Comment (jorgepiloto) [reply .github/workflows/ci_cd.yml:97]: I applied this suggestion to all the files with this typo.

File: .gitignore
Comment (jorgepiloto) [reply .gitignore:100]: Not really... This is the default `.gitignore` file that we use for our Python projects. We usually leave as it is.

File: .gitignore
Comment (jorgepiloto) [reply .gitignore]: I'm adding a clarification on this.

**Pull request merged**
